roles/cloud_connector/tasks/main.yml

```diff
@@ -0,0 +1,54 @@
+- name: Install rhc
+  block:
+    - name: Check if foreman-maintain installed
+      ansible.builtin.package_facts:
+        manager: auto
+
+    - name: Unlock packages
+      command: "foreman-maintain packages unlock"
+      when: "'foreman-maintain' in ansible_facts.packages"
+
+    - name: Install yggdrasil-worker-forwarder
+      package:
+        name: yggdrasil-worker-forwarder
+        state: present
+
+    - name: Install rhc
+      package:
+        name: rhc
+        state: present
+
+    - name: Lock packages
+      command: "foreman-maintain packages lock"
+      when: "'foreman-maintain' in ansible_facts.packages"
+
+- name: Create workers directory
+  file:
+    state: directory
+    path: "/etc/rhc/workers"
+    owner: root
+    group: root
+    mode: 755
+
+- name: Configure rhc-cloud-connector-worker
+  ansible.builtin.template:
+    src: "rhc-worker-cloud-connector.toml.j2"
+    dest: "{{ cloud_connector_config_file }}"
+    owner: root
+    group: root
+    mode: 640
+
+- name: Create rhcd worker
+  ansible.builtin.template:
+    src: rhc-cloud-connector-worker.j2
+    dest: /usr/libexec/rhc/rhc-cloud-connector-worker
+    owner: root
+    group: root
+    mode: 0755
+
+- name: Ensure rhcd started
+  service:
+    name: rhcd
+    state: started
+
+- include: cleanup_receptor.yml
```

Comment:
We are missing the RHD id reporting. We can use this API to report it back to Foreman: https://apidocs.theforeman.org/foreman/3.2/apidoc/v2/settings/update.html

Reply:
I have added this at the bottom of this set of tasks -- take a look at the logic and let me know your thoughts.